Add code of conduct for the team (#3)
alan-turing-institute/open-community-building: CODE_OF_CONDUCT.md

reported to the community leaders responsible for enforcement.
If noticed by community leaders, incidents should be recorded and shared with one or more 
members of the Open Research Community Building Team for transparency.

All reports will be reviewed and investigated promptly and fairly.

All community leaders are obligated to respect the privacy and security of the
reporter of any incident.
Together we strive to foster an environment where everyone feels valued, respected, and included.

## 13. Reporting Incident

**13.1** Code of Conduct Contact for Reporting

If you experience or witness unacceptable behaviour, or have any other concerns, please report it 
by contacting the Team Lead, Malvika Sharan (email: msharan@turing.ac.uk). Malvika may involve
the Tools, Practices and Systems Programme Director, Kirstie Whitaker when incidents are perceived 
as difficult (not straightforward) except when any potential conflict of interest may arise.

To report an issue involving the team lead, please email directly to the Tools, Practices and Systems 
Programme Director, Kirstie Whitaker. Since she is currently on maternity leave, please contact TPS

> Programme Manager Arielle Bennett (abennett@turing.ac.uk) - who is also our ombudsperson. 

Author: @arielle-bennett thanks for agreeing to be added in the CoC. Can you please look at the CoC and the details that this role entails?

Author: Approved by Arielle personally.

behaviour was inappropriate.
1. an apology to the person harmed or the team/community as a whole - this will be done via the person handling the report
1. an action plan for the reportee to improve their awareness and correct future behaviours
1. reportee will have no interaction with the reporter, mediating interaction between reporter and reportee is not in scope of this Code of Conduct
1. reportee will have no unsolicited interaction with reporter (in rare cases, when reporter is not anonymous) and those enforcing the Code of Conduct. This includes avoiding interactions in community spaces as well as external channels like social media.
1. not publishing the content (code, documents, media or other materials) that violated the Code of Conduct, until necessary changes have been made and agreed upon by the person handling the report.
1. An imposed suspension, such as asking reportee to “take a week or month off” from the GitHub repo or team activity.
1. recommended training activity relevant to correcting the behaviour or building awareness/understanding required to right the wrongs.
1. a stronger warning with consequences for continued behaviour. 

> 

Author: @arronlacey should we add something here to address what if a resolution can't be agreed (from #104).

Assumption is that people involved in report handling will arrive to some kind of resolution. However, we can include that if resolution is not reached, the ombudsperson will be cconsulted and any delay will be communicated with the reporter and reportee.

Author: ```suggestion
1. if a resolution is not reached, the ombudsperson will be consulted and any delay or status update for the CoC enforcement will be communicated with the reporter and reportee.
```

or unacceptable behaviour by other participants. 
We are committed to providing a welcoming, collaborative and inspiring experience 
for all members of this team and expect our code of conduct to be honoured.

## 2. Our Commitment

We as members, contributors, and leaders commit to make participation in our
community a harassment-free experience for everyone, regardless of age, body
size, visible or invisible disability, ethnicity, sex characteristics, gender
identity and expression, level of experience, education, socio-economic status,
nationality, personal appearance, race, religion, or sexual identity
and orientation.

We commit to act and interact in ways that contribute to an open, welcoming,
diverse, inclusive, and healthy community.

## 3. Restorative Practice

We want to establish a Code of Conduct that focuses on creating a welcoming and safe environment for
everyone involved.
If a participant does not abide by the Code of Conduct, we will use the 
“restorative practices” approach to handle the situation. Restorative Practices has 
their roots in “restorative justice,” a way of looking at wrongdoing that focuses on 

> repairing the harm done to people and the relationships rather than on punishing offenders or wrongdoer. 

Reviewer: ```suggestion
repairing the harm done to people and the relationships rather than on punishing offenders or wrongdoers. 
```

## 4. Our Standards

- 4.1 **Be friendly and patient.**

- 4.2 **Be welcoming**: we strive to build a welcoming and supportive team.
  As community facilitators, we also strive to apply this value in creating 

>   Communities of Practice that welcomes and support members of all backgrounds 

Reviewer: ```suggestion
  Communities of Practice that welcomes and supports members of all backgrounds 
```

Reviewer 2: Is it 'welcome and support' since communities of practice is plural?

  gender identity and expression, age, size, family status, political belief,
  religion, and mental and physical ability.

- 4.3 **Be considerate**: Your work will be used by other people, and you in turn
  will depend on the work of others. Any decision you take will affect users and
  colleagues, and you should take those consequences into account when making
  decisions. Remember that we're an open source project. This means that contributors
  might not be communicating in their primary language.

- 4.4 **Be respectful**: Not all of us will agree all the time, but disagreement is
  no excuse for poor behaviour and poor manners. We might all experience some
  frustration now and then, but we cannot allow that frustration to turn into a
  personal attack. It’s important to remember that a project where people feel
  uncomfortable or threatened is not a productive one.

- 4.5 **Be careful in the words that we choose**: We are
  professionals, and we conduct ourselves professionally. Be kind to others. Do
  not insult or put down other participants. Harassment and other exclusionary
  behaviour aren't acceptable. This includes, but is not limited to Violent

>   threats or language directed against another person, Discriminatory jokes and

Reviewer: ```suggestion
  threats or language directed against another person, discriminatory jokes and
```

>   language, Posting sexually explicit or violent material, Posting (or

Reviewer: ```suggestion
  language, posting sexually explicit or violent material, posting (or
```

  threatening to post) other people’s personally identifying information

>   (“doxing”), Personal insults, especially those using racist or sexist terms,

Reviewer: ```suggestion
  (“doxing”), personal insults, especially those using racist or sexist terms,
```

>   Unwelcome sexual attention, Advocating for or encouraging, any of the above

Reviewer: ```suggestion
  unwelcome sexual attention, advocating for or encouraging, any of the above
```

>   behaviour, and Repeated harassment of others. In general, if someone asks you to

Reviewer: ```suggestion
  behaviour, and repeated harassment of others. In general, if someone asks you to
```

based on any of the protected characteristics above, including participants with
disabilities.

## 7. Restorative Practice Statement and Principles

Restorative practices are based on the belief that human beings are happier, more 
cooperative and productive, and more likely to make positive changes in their behaviour 
when those in positions of authority do things with them, rather than to them or for them. 
The most critical function of restorative practices is restoring and building relationships. 
People in charge must meet with those they have wronged, explore what happened, 
and make necessary amends.

## 8. Restorative Practice Principles in this Code of Conduct

1. Focuses on harms rather than the intention of wrongdoers.
2. Prioritises persons affected and the consequent needs of victims and safety in communities.
3. Addresses obligations of the wrongdoer resulting from those harms, as well as the team/community’s obligations to both victims and wrongdoers.
4. Creates opportunity for all individuals in this team, who have a legitimate stake in a given situation, to be heard and their concerns addressed.

> 5. Seeks to right the wrongs that have been done so that victims feel safe and valued, and wrongdoers feel restored to the team/community

Reviewer: ```suggestion
5. Seeks to right the wrongs that have been done so that victims feel safe and valued, and wrongdoers feel restored to the team/community.
```

6. Uses inclusive, collaborative processes giving fair opportunity to right the wrongs.

## 9. Restorative Disciplines

1. Acknowledges that trust, safe space and relationships are central to community building.
2. Must establish policies and procedures that handle incident reports in a way that strengthens relationships.
3. Focuses on harms done rather than rules broken.
4. Gives voice to the person who has been harmed.

> 5. Engages in collaborative problem solving

Reviewer: ```suggestion
5. Engages in collaborative problem solving.
```

6. Empowers change and growth for all involved.

> 7. Enhances responsibility for actions and attitudes of all involved

Reviewer: ```suggestion
7. Enhances responsibility for actions and attitudes of all involved.
```

We will respect the confidentiality of reporter(s) all the time to protect individuals who have been harmed.
In case the reporter's identity can not be hidden (if easily identified through the details of the report), the person handling the Code of Conduct report will discuss the possible ways to protect reporters from future incidents.

## 14. Terminology

* **14.1** Reporter: Person reporting an incident.
* **14.2** Reportee: Person being reported.

## 15. Report Handling and Enforcement

After receiving a report, the community leaders (or person handling the report), team lead or programme director, in this case, will review the incident report and follow this process to determine the cause and consequences that violated this Code of Conduct and take further actions:

### **15.1** If the breach is considered within the scope of this Code of Conduct

Community leaders will reach out to the reporter to ask for further information if the submitted report is not complete or needs some additional context.
Community leaders will also ensure that person impacted or harmed by the reported incident is safe and identify how they can be further supported.

> **15.1.1** A meeting with the reportee will be scheduled where the person handling the report will discuss four questions:

Author: ```suggestion
**15.1.1** A meeting with the reportee will be scheduled where the person handling the report will discuss five questions:
```

- [Research Ethics Policy](https://mathison.turing.ac.uk/Interact/Pages/Content/Document.aspx?id=2241&SearchId=0&utm_source=interact&utm_medium=general_search&utm_term=code+of+conduct)
- [Data protection policy](https://mathison.turing.ac.uk/Interact/Pages/Content/Document.aspx?id=2184&SearchId=23865&utm_source=interact&utm_medium=general_search&utm_term=data+protection)

### **15.2.2** If the reportee is not the Turing Employee
- an appropriate authority or law enforcement mechanism will be identified who can respond to the reported incident with the appropriate action.

## 16. Resolutions for reports in the scope of this Code of Conduct

Here are examples of possible resolutions to a report. 
This list is not comprehensive, and any action necessary to reach a fair resolution will be taken as needed. 
Possible resolutions to an incident include:

1. a private communication from community leaders, providing
clarity around the nature of the violation and an explanation of why the
behaviour was inappropriate.

> 1. an apology to the person harmed or the team/community as a whole - this will be done via the person handling the report

Reviewer: ```suggestion
1. an apology to the person harmed or the team/community as a whole - this will be done via the person handling the report.
```

> 1. an action plan for the reportee to improve their awareness and correct future behaviours

Reviewer: ```suggestion
1. an action plan for the reportee to improve their awareness and correct future behaviours.
```

> 1. reportee will have no interaction with the reporter, mediating interaction between reporter and reportee is not in scope of this Code of Conduct

Reviewer: ```suggestion
1. reportee will have no interaction with the reporter, mediating interaction between reporter and reportee is not in scope of this Code of Conduct.
```

# 1. Code of Conduct for the Open Research Community Building Team

> _A code of conduct is a set of rules outlining the norms, rules, and responsibilities or proper practices of an individual party or an organization._

This Code of Conduct document outlines our expectations for 

> participants within the team as well as this online repository (project).

Reviewer 2: ```suggestion
participants within the team as well as this online repository (and the ideas, projects, and collaborations that might emerge from it).
```
Just adding this for clari

We also provide steps to reporting any concern, uncomfortable situation 
or unacceptable behaviour by other participants. 
We are committed to providing a welcoming, collaborative and inspiring experience 
for all members of this team and expect our code of conduct to be honoured.

## 2. Our Commitment

We as members, contributors, and leaders commit to make participation in our
community a harassment-free experience for everyone, regardless of age, body
size, visible or invisible disability, ethnicity, sex characteristics, gender
identity and expression, level of experience, education, socio-economic status,
nationality, personal appearance, race, religion, or sexual identity
and orientation.

We commit to act and interact in ways that contribute to an open, welcoming,
diverse, inclusive, and healthy community.

## 3. Restorative Practice

We want to establish a Code of Conduct that focuses on creating a welcoming and safe environment for
everyone involved.
If a participant does not abide by the Code of Conduct, we will use the 

> “restorative practices” approach to handle the situation. Restorative Practices has 

Reviewer 2: ```suggestion
“restorative practices” approach to handle the situation. Restorative practices have
```

their roots in “restorative justice,” a way of looking at wrongdoing that focuses on 

> repairing the harm done to people and the relationships rather than on punishing offenders or wrongdoer. 

Reviewer 2: ```suggestion
repairing the harm done to people and the relationships rather than on punishing offenders or wrongdoer. This approach emerged from studies of and in reaction to incarceration and punishment within criminal justice systems, and has its roots in indigenous belief systems.
```
I suggest this addition for greater context & intellectual lineage, especially for those that might not be familiar with the approach.

Author: Slight rewording suggestion for this context:
> Although the 'restorative justice' term was in 1970 in the context of social work and justice systems, restorative practice has its roots in indigenous practices employed in cultures all over the world.

  language, Posting sexually explicit or violent material, Posting (or
  threatening to post) other people’s personally identifying information
  (“doxing”), Personal insults, especially those using racist or sexist terms,
  Unwelcome sexual attention, Advocating for or encouraging, any of the above
  behaviour, and Repeated harassment of others. In general, if someone asks you to
  stop, then stop.

- 4.6 **Try to understand why we disagree**: Disagreements, both social and
  technical, happen all the time. We must resolve disagreements
  and differing views constructively. Remember that we’re different. Diversity
  contributes to the strength of our work, which is composed of people from
  a wide range of backgrounds. Different people have different perspectives on
  issues. Being unable to understand why someone holds a viewpoint doesn’t mean
  that they’re wrong. Don’t forget that it is human to err and blaming each
  other doesn’t get us anywhere. Instead, focus on helping to resolve issues and
  learning from mistakes.

## 5. Examples of expected and unacceptable behaviour

**5.1** Examples of behaviour that contributes to a positive environment for our
community include:

> 1. Demonstrating empathy and kindness toward other people

Reviewer 2: The numbers are displayed as a series of 1's, not sure why on my end.

Reviewer 3: This is a markdown "hack". If you look at the rendered copy, you'll see that the numbers progress as you'd expect. This means that authors don't have to keep track of how many bullets they have, and can also help mitigate issues where instructions might say "refer to step 3", but then a new PR adds a new step before so step 3 isn't step 3 anymore, etc.

Author: Thanks Sarah for addressing this!

Reviewer 2: Thanks so much @sgibson91 for the clarification! 👍

1. Being respectful of differing opinions, viewpoints, and experiences
1. Gracefully giving and accepting constructive feedback

> 1. Accepting responsibility for harm done to people/community, correcting our mistakes,

Reviewer 2: ```suggestion
1. Accepting responsibility for harm done to people and/or communities, correcting our mistakes,
```

>   and learning from the experience to avoid future harm

Reviewer 2: I wonder if it would be worth adding in quotes: 
```suggestion
  and learning from the experience to avoid future harm (following the 'regenerative justice' framework)
```

Author: I would avoid adding new concept unless strictly necessary. Has this term been mentioned before? If not, do we need to mention it at all?

Reviewer 2: I agree! Especially with the addition of the line above about regenerative justice, I don't think this is a necessary addition anymore.

- [Research Misconduct Policy](https://mathison.turing.ac.uk/Interact/Pages/Content/Document.aspx?id=2202&SearchId=0&utm_source=interact&utm_medium=general_search&utm_term=code+of+conduct)
- [Safeguarding policy](https://mathison.turing.ac.uk/Interact/Pages/Content/Document.aspx?id=2243&SearchId=0&utm_source=interact&utm_medium=general_search&utm_term=code+of+conduct)
- [Grievance procedure](https://mathison.turing.ac.uk/Interact/Pages/Content/Document.aspx?id=2217&SearchId=0&utm_source=interact&utm_medium=general_search&utm_term=code+of+conduct)
- [Wellbeing support services](https://mathison.turing.ac.uk/Interact/Pages/Content/Document.aspx?id=2228&SearchId=0&utm_source=interact&utm_medium=general_search&utm_term=code+of+conduct)
- [Research Ethics Policy](https://mathison.turing.ac.uk/Interact/Pages/Content/Document.aspx?id=2241&SearchId=0&utm_source=interact&utm_medium=general_search&utm_term=code+of+conduct)
- [Data protection policy](https://mathison.turing.ac.uk/Interact/Pages/Content/Document.aspx?id=2184&SearchId=23865&utm_source=interact&utm_medium=general_search&utm_term=data+protection)

> ### **15.2.2** If the reportee is not the Turing Employee

Author: ```suggestion
### **15.2.2** If the reportee is not a Turing Employee
```

1. an action plan for the reportee to improve their awareness and correct future behaviours.
1. reportee will have no unsolicited interaction with the reporter
1. reportee will have no unsolicited interaction with reporter (in rare cases, when reporter is not anonymous) and those enforcing the Code of Conduct. This includes avoiding interactions in community spaces as well as external channels like social media.
1. not publishing the content (code, documents, media or other materials) that violated the Code of Conduct, until necessary changes have been made and agreed upon by the person handling the report.
1. An imposed suspension, such as asking reportee to “take a week or month off” from the GitHub repo or team activity.
1. recommended training activity relevant to correcting the behaviour or building awareness/understanding required to right the wrongs.
1. a stronger warning with consequences for continued behaviour. 
1. if a resolution is not reached, the ombudsperson will be consulted and any delay or status update for the CoC enforcement will be communicated with the reporter and reportee.

> **16.1** After the Resolution is Agreed Upon

Author: ```suggestion

**16.1** After the Resolution is Agreed Upon
```

identity and expression, level of experience, education, socio-economic status,
nationality, personal appearance, race, religion, or sexual identity
and orientation.

We commit to act and interact in ways that contribute to an open, welcoming,
diverse, inclusive, and healthy community.

## 3. Restorative Practice

We want to establish a Code of Conduct that focuses on creating a welcoming and safe environment for
everyone involved.
If a participant does not abide by the Code of Conduct, we will use the 
“restorative practices” approach to handle the situation. Restorative Practices has 
their roots in “restorative justice,” a way of looking at wrongdoing that focuses on 
repairing the harm done to people and the relationships rather than on punishing offenders or wrongdoers. 

## 4. Our Standards

- 4.1 **Be friendly and patient.**

- 4.2 **Be welcoming**: we strive to build a welcoming and supportive team.
  As community facilitators, we also strive to apply this value in creating 
  Communities of Practice that welcomes and supports members of all backgrounds 
  and identities. This includes, but is not limited to

>   members of any race, ethnicity, culture, national origin, colour, immigration

Author: The 'commitment' section is supposed to be exhaustive. This one is not exhaustive, instead we highlight the top protected characteristics. This can be kept identical if you suggest.

>   members of any race, ethnicity, culture, national origin, colour, immigration

Reviewer 4: No, that's okay! I would maybe just remove "colour" from this list as I think it doesn't add anything to race and reads as perhaps older terminology? Do let me know if I'm missing something though 😊

>   members of any race, ethnicity, culture, national origin, colour, immigration

Author: Now that inconsistency is definitely bothering me. Let me suggest some consistent change.

>   members of any race, ethnicity, culture, national origin, colour, immigration

Author: I have merged "our commitment" with "diversity statement" into single clause 2: https://github.com/alan-turing-institute/open-community-building/blob/code-of-conduct/CODE_OF_CONDUCT.md#2-our-commitment-and-diversity-statement

To specifically address the inconsistency issue, I have now updated 4.2 that references clause 2.
Let me know if this actually works better.

>   members of any race, ethnicity, culture, national origin, colour, immigration

Reviewer 4: I like that!

>   members of any race, ethnicity, culture, national origin, colour, immigration

Author: Great! Thank you Eirini.

cooperative and productive, and more likely to make positive changes in their behaviour 
when those in positions of authority do things with them, rather than to them or for them. 
The most critical function of restorative practices is restoring and building relationships. 
People in charge must meet with those they have wronged, explore what happened, 
and make necessary amends.

## 8. Restorative Practice Principles in this Code of Conduct

1. Focuses on harms rather than the intention of wrongdoers.
2. Prioritises persons affected and the consequent needs of victims and safety in communities.
3. Addresses obligations of the wrongdoer resulting from those harms, as well as the team/community’s obligations to both victims and wrongdoers.
4. Creates opportunity for all individuals in this team, who have a legitimate stake in a given situation, to be heard and their concerns addressed.
5. Seeks to right the wrongs that have been done so that victims feel safe and valued, and wrongdoers feel restored to the team/community.
6. Uses inclusive, collaborative processes giving fair opportunity to right the wrongs.

## 9. Restorative Disciplines

> 1. Acknowledges that trust, safe space and relationships are central to community building.

Reviewer 3: ```suggestion
1. Acknowledges that trust, safe spaces and relationships are central to community building.
```

?

**15.1.2** When the incident in question inflicted harm on another person, additional 
questions will be asked on behalf of the person(s) harmed:
- What did you think when you realized what had happened?
- What impact has this incident had on you and others?
- What do you think needs to happen to make things right?

The person handling the report may follow up with reportee or reporter after this discussion 
with any additional questions to help them reach a resolution.
As per 13.3, mediating interaction or communication between reporter and reportee is not in scope of this Code of Conduct.

### **15.2** If the breach is considered outside the scope of this Code of Conduct

### **15.2.1** If the reportee is an employee of the Turing 

Code of Conduct should be considered as a norm-setting mechanism for the Open Research Community Building Team before any involvement of HR in the interactions that take place within the team. 

If the reportee is a Turing employee, for instance, Community Managers and senior researchers, and the incident is considered out of the scope of this Code of Conduct due to the severity of the matter/case, an HR process (formal or informal), will be started.

> In this case, the report will be passed to the HR and/or appropriate personnel under the Turing policies and their policy related enforcement mechanism will take precedent over the Code of Conduct enforcement.

Author: ```suggestion
In this case, the report will be passed to the HR and/or appropriate personnel under the Turing policies and their policy related enforcement mechanism will take precedent over this Code of Conduct enforcement.
```

team a supportive and harassment-free experience for everyone.
We commit to act and interact in ways that contribute to an open, welcoming,
diverse, inclusive, and healthy team, as well as the communities we facilitate.
We encourage everyone in the team to participate in all team activities and work.

Although this list cannot be exhaustive, we explicitly honour diversity in age,
gender, gender identity or expression, culture, ethnicity, language, national
origin, political beliefs, profession, race, religion, sexual orientation,
socioeconomic status, and technical ability. We will not tolerate discrimination
based on any of the protected characteristics above, including participants with
visible or invisible disabilities.

We are committed to facilitating Communities of Practice that are connected with 
each others' work. Although we will fail at times, we seek to treat everyone both as
fairly and equally as possible. Whenever a member has made a mistake, we
expect them to take responsibility for it. If someone has been harmed or
offended, it is our responsibility to listen carefully and respectfully, and do

> our best to right the wrong.

Author: ```suggestion
our best to right the wrong.

Although the 'restorative justice' term was coined in 1970 in the context of social work and justice systems, restorative practice has its roots in indigenous practices employed in cultures all over the world.
```

Reviewer 2: Looks great!
